File: app/src/main/java/com/example/mealna/MainActivity.java
package com.example.mealna;

import android.content.Intent;
import android.os.Bundle;
import android.os.Handler;
import android.os.Looper;

import androidx.activity.EdgeToEdge;
import androidx.appcompat.app.AppCompatActivity;
import androidx.core.graphics.Insets;
import androidx.core.view.ViewCompat;
import androidx.core.view.WindowInsetsCompat;

import java.util.Objects;

public class MainActivity extends AppCompatActivity {
    Handler handler;
    Intent intent;
    private Compiler EdgeToEdge;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        EdgeToEdge.enable();
        setContentView(R.layout.activity_main);
        ViewCompat.setOnApplyWindowInsetsListener(findViewById(R.id.main), (v, insets) -> {
            Insets systemBars = insets.getInsets(WindowInsetsCompat.Type.systemBars());
            v.setPadding(systemBars.left, systemBars.top, systemBars.right, systemBars.bottom);
            return insets;
        });

        // hide action bar
        Objects.requireNonNull(getSupportActionBar()).hide();

        // loading screen
        handler = new Handler(Looper.getMainLooper());
        handler.postDelayed(() -> {
                intent = new Intent(MainActivity.this, HomeActivity.class);
                startActivity(intent);
                finish();
        }, 3000);
    }
}

File: app/src/main/java/com/example/mealna/MainActivity.kt
package com.example.mealna

import android.os.Bundle
import android.util.Log
import androidx.activity.ComponentActivity
import com.google.firebase.FirebaseApp

class MainActivity : ComponentActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)

        val app = FirebaseApp.initializeApp(this)
        if (app != null) {
            Log.d("FIREBASE", "Firebase BERHASIL TERHUBUNG!")
        } else {
            Log.e("FIREBASE", "Firebase GAGAL TERHUBUNG!")
        }
    }
}
